File: backend/src/main/java/iliev/yt/share/backend/message/dto/MessageOutputDto.java
package iliev.yt.share.backend.message.dto;

import iliev.yt.share.backend.chat.dto.ChatOutputDto;
import iliev.yt.share.backend.message.enums.DeliveryStatus;
import iliev.yt.share.backend.user.dto.UserOutputDto;
import java.util.UUID;

public record MessageOutputDto(
        UUID id,
        String content,
        DeliveryStatus status,
        ChatOutputDto chat,
        UserOutputDto sender
) {
}

File: YTShare.Android/app/src/main/java/iliev/yt/share/mobile/data/remote/dto/MessageOutputDto.kt
package iliev.yt.share.mobile.data.remote.dto

import kotlinx.serialization.Serializable

@Serializable
data class MessageOutputDto(
    val id: String,
    val content: String,
    val status: String,
    val chat: ChatOutputDto,
    val sender: UserOutputDto,
    val createdAt: String
)
